generator/svelte/src/lib/utils.ts

```diff
@@ -40,15 +41,82 @@ export const isOlderThanAWeek = (isoDate: string) => {
 	return dayjs.utc(isoDate).isBefore(dayjs.utc().subtract(7, 'day'), 'day');
 };
 
+export const getCurrentLocalDateTime = () => {
+	return dayjs.utc().local();
+};
+
 /** Data transformation */
 
-export const extractPrayersFromMasjids = (masjids: [string, IMasjid][]) => {
-	if (!masjids.length) return [];
-	const masjid = masjids[0][1];
-	if (!masjid) return [];
+export const getPrayers = () => {
+	const prayers: Partial<TPrayer>[] = PRAYER_NAMES.map((prayer) => ({ name: prayer }));
+	const tPrayers: TPrayer[] = [];
+
+	for (let i = 0; i < prayers.length; i++) {
+		const prayer = prayers[i];
+		const next = prayers[i + 1] || prayers[0];
+		prayer.next = next as TPrayer;
+		tPrayers.push(prayer as TPrayer);
+	}
+
+	return tPrayers;
+};
+
+export const getNextPrayerForMasjid = (masjid: { iqamas: Record<string, { time: string }> }) => {
+	const currentTime = getCurrentLocalDateTime();
+	const prayers = getPrayers();
+
+	const nextPrayer = prayers.find((prayer) => {
+		const prayerTime = masjid.iqamas[prayer.name]?.time.toUpperCase();
+		const prayerDateTime = dayjs(prayerTime, 'h:mm A').local();
```

Comment:
Is this why Isha in Shalimar shows at the top (08:15)?

Reply:
It's because of sorting, masjids are sorted starting from lowest time (ASC), I believe the sorting algorithm doesn't parse the time, it's only a string comparison. Shalimar starts with 0 (08:15 PM) that's why it shows on the top.

Reply:
This will fire back in some cases. 1 will come before 12. I will create a new issue to parse the times on the backend so that they become properly sortable. It will also help identify the next prayer more reliably.

Reply:
Created #67.